### PR TITLE
azure-cli: update cffi resource

### PR DESCRIPTION
**Pull request by carlocab**
Bug report: https://github.com/Homebrew/brew/issues/10152#issuecomment-753334171

I tried switching the URL to PyPI and using `update-python-resources` to update everything, but that lead to a build failure. The `install` block probably needs to be updated first before that works. In the meantime, this should at least fix the reported problem.

-----

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

**Review by Rylan12 — APPROVED** [review]
We should definitely try to transfer to the PyPI source so that issues like this can (hopefully) be avoided in the future. Especially for these formulae with _huge_ numbers of resources.

**Comment by carlocab** [comment]
Yep, sorry for not waiting for feedback on this one. However, the user who reported the bug confirmed that this fixed it, so I figured it was safe to merge.

I thought about fixing the install block to work with the PyPI URL, but:
1. that would've delayed the fix; and,
2. I was worried about breaking Azure CLI's CI, as they seem to have automated updates for it: https://github.com/Homebrew/homebrew-core/pull/67930

**Comment by carlocab** [comment]
Followed up on the PyPI URL migration here: https://github.com/Azure/azure-cli/issues/16388

**Comment by Rylan12** [comment]
No worries. I was going to trigger the merge after I approved anyway.

I wouldn't worry at all about breaking their CI. We should do what's best for Homebrew. If their stuff doesn't work, we can just fix the branch manually or close and open our own PR.

**Pull request closed without merging**
